.github/workflows/test.yml

```diff
@@ -1,323 +1,85 @@
-                    "tests/end_to_end/candid_rpc/functional_syntax/stable_b_tree_map_instruction_threshold",
-                    "tests/end_to_end/candid_rpc/functional_syntax/stable_structures",
-                    "tests/end_to_end/candid_rpc/functional_syntax/timers",
-                    "tests/end_to_end/candid_rpc/functional_syntax/tuple_types",
-                    "tests/end_to_end/candid_rpc/functional_syntax/update",
-                    "tests/end_to_end/candid_rpc/functional_syntax/vanilla_js",
-                    "tests/end_to_end/http_server/apollo_server",
-                    "tests/end_to_end/http_server/audio_and_video",
-                    "tests/end_to_end/http_server/autoreload",
-                    "tests/end_to_end/http_server/bitcoinjs_lib",
-                    "tests/end_to_end/http_server/bitcore_lib",
-                    "tests/end_to_end/http_server/ethers",
-                    "tests/end_to_end/http_server/express",
-                    "tests/end_to_end/http_server/fetch_ic",
-                    "tests/end_to_end/http_server/file_protocol",
-                    "tests/end_to_end/http_server/fs",
-                    "tests/end_to_end/http_server/hybrid_canister",
-                    "tests/end_to_end/http_server/internet_identity",
-                    "tests/end_to_end/http_server/large_files",
-                    "tests/end_to_end/http_server/nest",
-                    "tests/end_to_end/http_server/new",
-                    "tests/end_to_end/http_server/open_value_sharing",
-                    "tests/end_to_end/http_server/sqlite",
-                    "tests/end_to_end/http_server/sqlite_drizzle",
-                    "tests/end_to_end/http_server/sqlite_typeorm",
-                    "tests/end_to_end/http_server/tfjs",
-                    "tests/end_to_end/http_server/web_assembly",
-                    "tests/property/ic_api/caller",
-                    "tests/property/ic_api/chunk",
-                    "tests/property/ic_api/id",
-                    "tests/property/ic_api/instruction_counter",
-                    "tests/property/ic_api/is_controller",
-                    "tests/property/ic_api/performance_counter",
-                    "tests/property/ic_api/time",
-                    "tests/property/ic_api/trap"
-                  ]
-                  END
-                  )
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//'%'/'%25'}"
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//$'\n'/'%0A'}"
-                  EXAMPLE_DIRECTORIES="${EXAMPLE_DIRECTORIES//$'\r'/'%0D'}"
-                  echo "::set-output name=example_directories::$EXAMPLE_DIRECTORIES"
-            - id: should_run_tests
+                  directories: |
+                      ./examples
+                      ./tests
+                  exclude_dirs: |
+                      ${{ env.IS_FEATURE == 'true' && format('
```

Comment:
Is there no more elegant way to do this than bash? Could we somehow do it with just yaml variables and if statements or something?

Reply:
The current approach uses inline Bash to dynamically set exclude_dirs, which I believe keeps the logic compact and flexible within a single job. Alternatives like splitting into multiple jobs or using external files could be considered, but they might add unnecessary complexity. However, I'm open to refining this if you feel strongly about it!